ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -526,16 +585,17 @@ impl<'a> Assemble<'a> {
                     ],
                 );
                 debug_assert_eq!(_rhs_reg, Rq::RCX);
-                match size {
-                    1 => dynasm!(self.asm; shr Rb(lhs_reg.code()), cl),
-                    2 => dynasm!(self.asm; shr Rw(lhs_reg.code()), cl),
-                    4 => dynasm!(self.asm; shr Rd(lhs_reg.code()), cl),
-                    8 => dynasm!(self.asm; shr Rq(lhs_reg.code()), cl),
+                match bit_size {
+                    1..=64 => {
+                        // Ensure we shift in zeros at the most-significant bits.
```

Comment:
I eventually worked out what this means (I think), but it confused me for a while. I think it means "For the `shr` to work correctly, we need to ensure there are no undefined bits, as they would end up appearing in the defined part of the register".

Reply:
Yes, but I was trying to keep the comments snappy as it's a recurring theme and we don't want to duplicate a lengthy explanation every time the theme reoccurs.